### ISSUE TITLE
Anyone try to enable file vault? Having weird issues

### ISSUE THREAD

**Issue by dude2k5** [ameeno/Dell-E7440-Hackintosh#31]
This guide saved me lots of time, did all the hard work, TY very much

I can boot in fine, everything works. I just wanted to put file vault on to be more safe/encrypted

When you enable it, wait for it to finish fully. Then reboot. When you reboot, It boots without showing the OS, but that's because you need to unhide the preboot drives, then it will boot (go into recovery, mount the efi partition, go into the config file and edit the hidden tag to erase preboot (the whole line), and leave recovery alone

Now, the weird thing is, it needs me to keep tapping the keyboard while showing the apple logo all the way until i put my password. after, then it works perfectly again. However, until i get past the login screen, i need to tap like crazy for it to "load". If I do nothing, it will sit for minutes doing nothing. Any ideas why?

**Comment by ameeno** [comment]
No idea pal, I dont WRITE or make the kexts, or the file vault hacks or the clover that makes all this Hackintosh stuff possible.

there are far greater minds out there than me, I have just bundled an install pack that works for this model of laptop that I have had success with.

I suspect that as your drive is encrypted, perhaps boot cache is not saved or readable, and is therefore rebuilt on startup? this could conceivably cause a delay?

Perhaps there is a hidden prompt you cant see because clover is designed to show you an apple logo (even showing the apple logo is basically a hack)

I would always edit the clover config file to add -v to the boot options for debugging until I am done and entirely happy with my install.

the boot flag -v is for verbose mode and will display all the bootup text spam allowing you to SEE what is going on and what the machine is actually doing at boot time.

Honestly, I do not recommend using file vault or disk-level encryption on a Hackintosh, too many variables can break.

This is your boot drive and boot OS, you should be glad it even works at all on unsupported hardware (thanks to the brilliant minds that built these mods in the first place!!!!)

now, I can appreciate you want to have encrypted files and folders, for that I recommend perhaps something like TrueCrypt or encrypted folders INSIDE your computer, that can unlock folders after loading up.

but to Encrypt the entire OS drive (which in layman's terms, uses keys to encrypt and decrypt data on the fly) is risky.

you do realise a Hackintosh is a cloned/emulated SMC hardware and an EFI bootloader to set custom flags right???

The real SMC does not exist on your Hackintosh. Things apple relies upon to encrypt and decrypt data do not exist.

the kexts you run via your emulated SMC's are either reverse-engineered or ripped from original macOS files.

Now if you're experimenting that is fine, I guess.

and if you intend to contribute to the community with patches, where you should be looking is boot logs and debug logs (You can press f1 on the clover page and then f3 and f4 I think to dump logs to your EFI)

and you could use verbose mode to see what's going on under the hood.

But you need to accept that a Hackintosh is a TOY. I would not rely on it for essential things or mission-critical work.

if you need a fully encrypted OS or disk-level encryption, use arch or ubuntu + luks containers, or perhaps go as far as tails of a USB.

you shouldn't be using a Hackintosh and trying to get disk-level encryption on it, its a ticking timebomb in my humble opinion.


Thanks for your feedback none the less, and I'm happy my repo helped you make progress.

**Comment by ameeno** [comment]
Actually,

truecrypt id discontinued, apparently, https://alternativeto.net/software/veracrypt/about/ is the replacement

**Comment by dude2k5** [comment]
Def appreciate all the gathering, it helped a lot. 

Turned out I just needed to do this, so if anyone else has similar try it: https://www.insanelymac.com/forum/topic/340805-solved-filevault-take-much-time-to-ask-for-password/


Also, yup, It's def a toy. This laptop has 3 OS now. All encrypted. I run Win/Mint/Mac on it, it's nice to be able to use any. 

I just dislike leaving stuff "open" persay. 

Ty again.

**Comment by ameeno** [comment]
Great so you switched from fakesmc to vsmc?

**Comment by dude2k5** [comment]
Yeah, these are the files in uefi and drivers64uefi

![Screen Shot 2019-11-08 at 3 48 30 PM](https://user-images.githubusercontent.com/24706365/68518223-d8b29980-023f-11ea-8bef-05223d7602b0.png)
![Screen Shot 2019-11-08 at 3 47 00 PM](https://user-images.githubusercontent.com/24706365/68518226-dbad8a00-023f-11ea-96af-6d1f5e0e108b.png)
